src/utils/build.js

```diff
@@ -321,171 +315,158 @@ const build = (zipPath, sourceZipPath, wdir) => {
-        });
-      }
+  maybeNotifyAboutOutdated();
 
-      // No need to mention specifically we're validating style checks as that's
-      //   implied from `zapier validate`, though it happens as a separate process
+  // make sure our directories are there
+  await ensureDir(tmpDir);
+  await ensureDir(constants.BUILD_DIR);
 
-      return callAPI('/style-check', {
-        skipDeployKey: true,
-        method: 'POST',
-        body: rawDefinition
-      });
-    })
-    .then(styleChecksResponse => {
-      const errors = styleChecksResponse.errors;
-      if (!_.isEmpty(errors)) {
-        if (global.argOpts.debug) {
-          console.log('\nErrors:');
-          console.log(errors);
-          console.log('');
-        }
+  startSpinner('Copying project to temp directory');
+  await copyDir(wdir, tmpDir, {
+    filter: process.env.SKIP_NPM_INSTALL
+      ? dir => !dir.includes('.zip')
+      : undefined
+  });
 
-        throw new Error(
-          'We hit some validation errors, try running `zapier validate` to see them!'
-        );
-      }
+  let output = {};
+  if (!process.env.SKIP_NPM_INSTALL) {
+    endSpinner();
+    startSpinner('Installing project dependencies');
+    output = await runCommand('npm', ['install', '--production'], {
+      cwd: tmpDir
+    });
+  }
+
+  // `npm install` may fail silently without returning a non-zero exit code, need to check further here
+  const corePath = path.join(
+    tmpDir,
+    'node_modules',
+    constants.PLATFORM_PACKAGE
+  );
+  if (!await fs.exists(corePath)) {
+    throw new Error(
+      'Could not install dependencies properly. Error log:\n' + output.stderr
+    );
+  }
+  endSpinner();
+
+  startSpinner('Applying entry point file');
+  // TODO: should this routine for include exist elsewhere?
+  const zapierWrapperBuf = await readFile(
+    path.join(
+      tmpDir,
+      'node_modules',
+      constants.PLATFORM_PACKAGE,
+      'include',
+      'zapierwrapper.js'
+    )
+  );
+  await writeFile(`${tmpDir}/zapierwrapper.js`, zapierWrapperBuf.toString());
+  endSpinner();
+
+  startSpinner('Building app definition.json');
+  const rawDefinition = (await _appCommandZapierWrapper(tmpDir, {
+    command: 'definition'
+  })).results;
 
-      endSpinner();
-      startSpinner('Zipping project and dependencies');
-      return makeZip(tmpDir, wdir + path.sep + zipPath);
-    })
-    .then(() => {
-      return makeSourceZip(tmpDir, wdir + path.sep + sourceZipPath);
-    })
-    .then(() => {
-      // tries to do a reproducible build at least
-      // https://blog.pivotal.io/labs/labs/barriers-deterministic-reproducible-zip-files
-      // https://reproducible-builds.org/tools/ or strip-nondeterminism
-      endSpinner();
-      startSpinner('Testing build');
-
-      if (isWindows()) {
-        return {}; // TODO err, what should we do on windows?
+  const fileWriteError = await writeFile(
+    `${tmpDir}/definition.json`,
+    prettyJSONstringify(rawDefinition.results)
```

Comment:
I think you meant to write

```suggestion
    prettyJSONstringify(rawDefinition)
```

because `.results` has been done previously.